### PR TITLE
Add Semestry branch naming convention to checks

### PR DESCRIPTION
**Pull request by andrewmitchellsemestry**
Ref. No reference number

## Proposed changes

When creating PRs in TermTime, we keep getting messages to say that our branches don't match the naming conventions. To combat this, I propose adding our branch naming conventions to this repo so that we no longer have to ignore these warning and they will only show when needed.

## Testing

None

## Documentation
Updated the README

### CODE REVIEW

**Comment by MikeN123** [comment]
@johanwortel I believe https://semestry.atlassian.net/wiki/spaces/TT/pages/2737242113/Git+workflow#Branching-model discusses the TermTime branching model.

File: action.yml [hunk 1]
```diff
@@ -9,6 +9,14 @@ inputs:
     description: 'Development branch pattern'
     required: true
     default: '^(dev|develop|development)$'
+  semestry_staging_branch_pattern:
+    description: 'Semestry staging branch pattern'
+    required: true
+    default: '^([0-9]+\.[0-9]+-staging|[0-9]+\.[0-9]+\.[0-9]+-staging)$'
+  semestry_testing_branch_pattern:
```
Comment (johanwortel) [inline action.yml:16]: You could combine the two expressions into one with the third number in parenthesis `(\.[0-9]+)?`.  Or even take the second and third number in one expression. This is a matter of personal preference I think.

File: action.yml [hunk 2]
```diff
@@ -9,6 +9,14 @@ inputs:
...
+    description: 'Semestry Testing branch pattern'
+    required: true
+    default: '^([0-9]+\.[0-9]-testing|[0-9]+\.[0-9]+\.[0-9]+-testing)$'
   feature_branch_prefix:
```
Comment (johanwortel) [inline action.yml:20]: Forgot the + in the second number?

File: index.js [hunk 3]
```diff
@@ -56,6 +58,22 @@ async function checkTargetBranch() {
...
+            let msg = "⛔ Pull requests for fix branches should target a version-test branch.";
+            console.log(msg);
+            await postCommentIfBaseChanged(msg);
+        }
+    } else if (isStagingBranch(head)) {
+        if (isSemestryStagingBranch(base)) {
+            console.log("Head branch is a staging branch and targets a Semestry version-staging branch.");
+        } else {
+            let msg = "⛔ Pull requests for staging branches should target a staging-test branch.";
+            console.log(msg);
```
Comment (johanwortel) [inline index.js:74]: A `staging-test` branch? This terminology is not on the Wiki.

**Review by johanwortel — COMMENTED** [review]
Nothing is done with `development-test` and `development-staging`. Is that correct? Surely development is done on future versions?

File: action.yml (hunk 1 above)
Comment (andrewmitchellsemestry) [reply action.yml:16]: Yes I agree, the branch regex could be a lot more concise. I have updated this now

File: index.js (hunk 3 above)
Comment (andrewmitchellsemestry) [reply index.js:74]: Well observed, I have accidentally named this the wrong way

**Comment by andrewmitchellsemestry** [comment]
@johanwortel I have updated the PR so that the regex is more concise and will consider the `development-test` and `development-staging` branches

**Review by johanwortel — APPROVED** [review]

**Comment by MikeN123** [comment]
@johanwortel @andrewmitchellsemestry who will create a `v1.2.0` tag and update the `v1` tag?

**Comment by andrewmitchellsemestry** [comment]
@MikeN123 I have added the new `v1.2.0` tag and updated the `v1` tag to be linked with the latest changes.

**Pull request merged**